ckanext/recombinant/datatypes.py

```diff
@@ -96,9 +96,9 @@ def canonicalize(
         # User has overridden Excel format string, probably adding currency
         # markers or digit group separators (e.g.,fr-CA uses 1$ (not $1)).
         # Accept only "DDDDD.DD", discard other characters
-        canon = re.sub(r'[$,\s]', '', str("{:.2f}".format(dirty)))
+        canon = re.sub(r'[$,\s]', '', "{:.2f}".format(float(dirty)))
```

Comment:
@RabiaSajjad as the value can have a "$" string to it, you could do the `re.sub` THEN float that, and format as a `:.2f`.

I think that might work?
```
canon = "{:.2f}".format(float(re.sub(r'[$,\s]', '', str(dirty))))
```

I would just make sure that it works with commas, e.g. `$ 10,000,000`

Reply:
Also make sure that it works with `None` value.